frontend/src/components/ArtifactInfo.tsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as React from 'react';
+import {stylesheet} from 'typestyle';
+import {color, commonCss} from '../Css';
+import {formatDateString, getArtifactProperty} from '../lib/Utils';
+import {MlMetadataArtifact} from '../apis/service';
+import {ArtifactProperties, ArtifactCustomProperties} from '../lib/Api';
+import {ModelInfo} from './ModelInfo';
+
+export const css = stylesheet({
+  artifactInfo: {
+    display: 'flex',
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+  },
+  field: {
+    flexBasis: '300px',
+    marginBottom: '32px',
+  },
+  term: {
+    color: color.grey,
+    fontSize: '12px',
+    letterSpacing: '0.2px',
+    lineHeight: '16px',
+  },
+  value: {
+    color: color.secondaryText,
+    fontSize: '14px',
+    letterSpacing: '0.2px',
+    lineHeight: '20px',
+  }
+});
+
+export interface ArtifactInfoProps {
+  typeName: string;
+  artifact: MlMetadataArtifact;
+}
+
+export class ArtifactInfo extends React.Component<ArtifactInfoProps, {}> {
+
+  public render(): JSX.Element {
+    const {artifact} = this.props;
+    const createTime = formatDateString(
+      getArtifactProperty(artifact, ArtifactProperties.CREATE_TIME) || '');
+    return (
+      <section>
+        <h1 className={commonCss.header}>{this.props.typeName} info</h1>
+        <dl className={css.artifactInfo}>
+          <div className={css.field}>
```

Comment:
Doesn't have to be done in this PR (or at all), but each of these looks similar enough to be generated by a function. The only things that are changing are the `term` and `value`

Reply:
Agreed. This whole thing is pretty generic right now so it could easily be done in a map.